### PR TITLE
Cudos 992: Implement upload and instantiate contract

### PR DESCRIPTION
**Pull request by aemil145**
Changes:

2 new contract functions - upload and instantiate
new global function getContractFromCodeId - gets contract object from uploaded code in the network
updated documentation
updated e2e tests, added test for interact.js sample script

### CODE REVIEW

**Review by tgntr — APPROVED** [review]

File: packages/blast-core/utilities/network-utils.js [hunk 1]
```diff
@@ -28,8 +28,14 @@ async function getContractInfo(nodeUrl, contractAddress) {
   }
 }
 
+async function getCodeDetails(nodeUrl, codeId) {
+  const client = await CosmWasmClient.connect(nodeUrl)
+  return await client.getCodeDetails(codeId)
```
Comment (maptuhec) [inline packages/blast-core/utilities/network-utils.js]: redundant await in return statement

File: packages/blast-core/lib/cudos-contract.js [hunk 2]
```diff
@@ -10,81 +10,172 @@ const {
...
+    signer: null, label: null, funds: null
+  }) {
+    if (!this.#isUploaded()) {
+      throw new BlastError('Cannot instantiate contract that is not uploaded. ' +
+        'Contract\'s code must exist on the network before instantiating')
     }
-    if (!options.label) {
-      options.label = this.#contractLabel
+    options.signer = options.signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    options.label = options.label ?? this.#label
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js]: fix the issue here as discussed

File: packages/blast-core/lib/cudos-contract.js [hunk 3]
```diff
@@ -10,81 +10,172 @@ const {
...
+  }) {
+    if (this.#isUploaded()) {
+      throw new BlastError(`Cannot deploy contract labeled ${this.#label}. Contract is already uploaded. ` +
+      'Only new contracts can be deployed. Use "instantiate" for uploaded contracts')
     }
-    const uploadTx = await this.#uploadContract()
-    const initTx = await this.#initContract(uploadTx.codeId, initMsg, this.#contractLabel, options.funds)
-    this.#contractAddress = initTx.contractAddress
+    options.signer = options.signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    options.label = options.label ?? this.#label
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js]: same

File: packages/blast-core/lib/cudos-contract.js [hunk 4]
```diff
@@ -10,81 +10,172 @@ const {
...
       codeId,
-      initMsg,
+      msg,
       label,
       instantiateFee,
       { funds: funds }
     )
   }
+
+  #isUploaded() {
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js:170]: this function is making a bit harder for me to read the code. The function is not for checking whether something is done but returning a strange value that is not expected by the name. Only one return statement in the function is not an improvement from my point of view

File: packages/blast-core/lib/cudos-contract.js [hunk 5]
```diff
@@ -10,81 +10,172 @@ const {
...
+    return this.#codeId
+  }
+
+  #isDeployed() {
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js:177]: this function is making a bit harder for me to read the code. The function is not for checking whether something is done but returning a strange value that is not expected by the name. Only one return statement in the function is not an improvement from my point of view

**Review by maptuhec — CHANGES_REQUESTED** [review]

File: packages/blast-core/lib/cudos-contract.js (hunk 2 above)
Comment (aemil145) [reply packages/blast-core/lib/cudos-contract.js]: I will only leave` this.#label `in the contract object so the user is able to `getContractFromAddress()` and check the label by `getLabel() `function. (The purpose of having `label` will be similar to `creator`) Let me know if you have other suggestions

File: packages/blast-core/utilities/network-utils.js (hunk 1 above)
Comment (aemil145) [reply packages/blast-core/utilities/network-utils.js]: fixed

File: packages/blast-core/lib/cudos-contract.js (hunk 4 above)
Comment (aemil145) [reply packages/blast-core/lib/cudos-contract.js:170]: fixed as discussed

File: packages/blast-core/lib/cudos-contract.js (hunk 5 above)
Comment (aemil145) [reply packages/blast-core/lib/cudos-contract.js:177]: fixed as discussed

**Review by maptuhec — APPROVED** [review]

**Review by tgntr — APPROVED** [review]

**Pull request merged**